src/azure-cli/azure/cli/command_modules/keyvault/_client_factory.py

```diff
@@ -3,23 +3,104 @@
-    from azure.cli.core.commands.client_factory import get_mgmt_service_client
-    from azure.cli.core.profiles import ResourceType
-    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_KEYVAULT)
 
+class Clients(str, Enum):
+    vaults = 'vaults'
+    private_endpoint_connections = 'private_endpoint_connections'
+    private_link_resources = 'private_link_resources'
+    managed_hsms = 'managed_hsms'
 
-def keyvault_client_vaults_factory(cli_ctx, _):
-    return keyvault_client_factory(cli_ctx).vaults
 
+OPERATIONS_NAME = {
+    Clients.vaults: 'VaultsOperations',
+    Clients.private_endpoint_connections: 'PrivateEndpointConnectionsOperations',
+    Clients.private_link_resources: 'PrivateLinkResourcesOperations',
+    Clients.managed_hsms: 'ManagedHsmsOperations'
+}
 
-def keyvault_client_private_endpoint_connections_factory(cli_ctx, _):
-    return keyvault_client_factory(cli_ctx).private_endpoint_connections
+KEYVAULT_TEMPLATE_STRINGS = {
+    ResourceType.MGMT_KEYVAULT:
+        'azure.mgmt.keyvault{api_version}.{module_name}#{class_name}{obj_name}',
+    ResourceType.MGMT_PRIVATE_KEYVAULT:
+        'azure.cli.command_modules.keyvault.vendored_sdks.azure_mgmt_keyvault{api_version}.'
+        '{module_name}#{class_name}{obj_name}',
+    ResourceType.DATA_KEYVAULT:
+        'azure.keyvault{api_version}.key_vault_client#{class_name}{obj_name}',
+    ResourceType.DATA_PRIVATE_KEYVAULT:
+        'azure.cli.command_modules.keyvault.vendored_sdks.azure_keyvault_t1{api_version}.'
+        'key_vault_client#{class_name}{obj_name}',
+    ResourceType.DATA_PRIVATE_KEYVAULT_T2:
+        'azure.cli.command_modules.keyvault.vendored_sdks.azure_keyvault_t2{api_version}.'
+        'key_vault_client#{class_name}{obj_name}'
+}
 
 
-def keyvault_client_private_link_resources_factory(cli_ctx, _):
-    return keyvault_client_factory(cli_ctx).private_link_resources
+def is_mgmt_plane(resource_type):
+    return resource_type in [ResourceType.MGMT_KEYVAULT, ResourceType.MGMT_PRIVATE_KEYVAULT]
+
+
+def get_operations_tmpl(resource_type, client_name):
+    class_name = OPERATIONS_NAME.get(client_name, '') if is_mgmt_plane(resource_type) else 'KeyVaultClient'
+    return KEYVAULT_TEMPLATE_STRINGS[resource_type].format(
+        api_version='',
+        module_name='operations',
+        class_name=class_name,
+        obj_name='.{}')
+
+
+def get_docs_tmpl(cli_ctx, resource_type, client_name, module_name='operations'):
+    api_version = '.v' + str(get_api_version(cli_ctx, resource_type)).replace('.', '_').replace('-', '_')
+    if is_mgmt_plane(resource_type):
+        class_name = OPERATIONS_NAME.get(client_name, '') + '.' if module_name == 'operations' else ''
+    else:
+        class_name = 'KeyVaultClient.'
+    return KEYVAULT_TEMPLATE_STRINGS[resource_type].format(
+        api_version=api_version,
+        module_name=module_name,
+        class_name=class_name,
+        obj_name='{}')
+
+
+def get_client_factory(resource_type, client_name=''):
```

Comment:
I feel maybe this part of code could be abstracted to a common class by _client_factory_vendor.py for future local development work.